rewrite-core/src/main/java/org/openrewrite/marker/RecipeSearchResult.java

```diff
@@ -1,47 +1,41 @@
-/*
- * Copyright 2020 the original author or authors.
- * <p>
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
- * <p>
- * https://www.apache.org/licenses/LICENSE-2.0
- * <p>
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
- */
-package org.openrewrite.marker;
-
-import lombok.*;
-import org.openrewrite.Incubating;
-import org.openrewrite.Recipe;
-import org.openrewrite.internal.lang.Nullable;
-
-/**
- * Mark any AST element with "paint". Used by search visitors to mark AST elements that
- * match the search criteria. By painting AST elements in a tree, search results can be
- * contextualized in the tree that they are found in.
- */
-@Incubating(since = "7.0.0")
-@EqualsAndHashCode
-@ToString
-@Getter
-public class RecipeSearchResult implements SearchResult {
-    @EqualsAndHashCode.Include
-    private final Recipe recipe;
-
-    @Nullable
-    private final String description;
-
-    public RecipeSearchResult(Recipe recipe, @Nullable String description) {
-        this.recipe = recipe;
-        this.description = description;
-    }
-
-    public RecipeSearchResult(Recipe recipe) {
-        this(recipe, null);
-    }
-}
+/*
+ * Copyright 2020 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.marker;
+
+import lombok.Data;
+import lombok.EqualsAndHashCode;
+import org.openrewrite.Incubating;
+import org.openrewrite.Recipe;
+import org.openrewrite.internal.lang.Nullable;
+
+import java.util.UUID;
+
+/**
+ * Used by search visitors to mark AST elements that match the search criteria. By marking AST elements in a tree,
+ * search results can be contextualized in the tree that they are found in.
+ */
+@Incubating(since = "7.0.0")
+@Data

```

Comment:
Can we leave the two constructors to make description optional?

Reply:
Under this paradigm I expect that you'd almost never directly use `RecipeSearchResult` and always subclass to something language-specific like `JavaSearchResult`. But sure, since we're not going all the way to making `RecipeSearchResult` `abstract` might as well have the description-optional constructors.

Reply:
ah, i see. forgot about the language specific markers.